tests/test_lit_server.py

```diff
@@ -119,3 +122,57 @@ def test_run():
     assert '{"output":16.0}' in output
     os.remove("client.py")
     process.kill()
+
+
+@pytest.mark.asyncio()
+async def test_stream(simple_stream_api):
+    server = LitServer(simple_stream_api, stream=True)
+    expected_output = "prompt=Hello World generated_output=LitServe is streaming output".lower().replace(" ", "")
+
+    async with LifespanManager(server.app) as manager, AsyncClient(app=manager.app, base_url="http://test") as ac:
+        response = await ac.post("/stream-predict", json={"prompt": "Hello World"}, timeout=10)
+        assert response.status_code == 200, "Check if server is running and the request format is valid."
+        assert response.text == expected_output, "Server returns input prompt and generated output which didn't match."
+
+
+class FakeStreamPipe:
+    i = 0
+
+    def send(self, item):
+        if isinstance(item, StopIteration):
+            raise StopIteration("exit loop")
+        assert item == f"{self.i}", "This streaming loop generates number from 0 to 9 which is sent via Pipe"
+        self.i += 1
+
+
+def test_streaming_loop(loop_args):
+    def fake_predict(inputs: str):
+        for i in range(10):
+            yield {"output": f"{i}"}
+
+    def fake_encode(output):
+        assert inspect.isgenerator(output), "predict function must be a generator in streaming mode"
+        for out in output:
+            yield out["output"]
+
+    fake_stream_api = MagicMock()
+    fake_stream_api.decode_request = MagicMock(side_effect=lambda x: x["prompt"])
+    fake_stream_api.predict = MagicMock(side_effect=fake_predict)
+    fake_stream_api.encode_response = MagicMock(side_effect=fake_encode)
+
+    _, requests_queue, request_buffer = loop_args
+    request_buffer = Manager().dict()
+    request_buffer[1] = {"prompt": "Hello"}, FakeStreamPipe()
+
+    with pytest.raises(StopIteration, match="exit loop"):
+        run_streaming_loop(fake_stream_api, requests_queue, request_buffer)
+
+    fake_stream_api.predict.assert_called_once_with("Hello")
+    fake_stream_api.encode_response.assert_called_once()
+
+
+def test_litapi_with_stream(simple_litapi, simple_stream_api):
+    with pytest.raises(
+        ValueError, match="Both lit_api.predict and lit_api.encode_response must be generators in streaming mode."
```

Comment:
Let's be more explicit

```suggestion
        ValueError, match="When `stream=True` both `lit_api.predict` and `lit_api.encode_response` must generate values using `yield`"
```

and add a simple example inline